src/main/java/io/github/ryotan/code/util/CodeEnumReflectionUtil.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * These codes are licensed under CC0.
+ *
+ * http://creativecommons.org/publicdomain/zero/1.0/deed
+ * http://creativecommons.org/publicdomain/zero/1.0/deed.ja
+ */
+
+package io.github.ryotan.code.util;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.AnnotatedElement;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.util.Optional;
+import java.util.function.Predicate;
+import java.util.stream.Stream;
+
+import io.github.ryotan.code.CodeEnum;
+import io.github.ryotan.code.CodeEnum.Filter;
+
+public final class CodeEnumReflectionUtil {
+    private CodeEnumReflectionUtil() {
+    }
+
+    public static boolean isValidCodeEnumClass(String aClass) {
+        try {
+            return isValidCodeEnumClass(Class.forName(aClass));
+        } catch (ClassNotFoundException e) {
+            return false;
+        }
+    }
+
+    public static boolean isValidCodeEnumClass(Class<?> aClass) {
+        if (!Enum.class.isAssignableFrom(aClass) || !CodeEnum.class.isAssignableFrom(aClass)) {
+            return false;
+        }
+
+        return Stream.of(aClass.getGenericInterfaces()).filter(ParameterizedType.class::isInstance).map(ParameterizedType.class::cast)
+                .filter(CodeEnumReflectionUtil::isCodeEnumType).anyMatch(type -> hasSameParameterizedType(type, aClass));
+    }
+
+    public static Class<? extends CodeEnum> getCodeEnumClass(String code) {
+        try {
+            return getCodeEnumClass(Class.forName(code));
+        } catch (ClassNotFoundException e) {
+            throw new IllegalArgumentException("Could not load a class for " + code + ". Check if " + code + " exists in classpath.");
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    public static Class<? extends CodeEnum> getCodeEnumClass(Class<?> code) {
+        if (!isValidCodeEnumClass(code)) {
+            throw new IllegalArgumentException(code.getName() + " is not a valid CodeEnum class. " +
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: ['+' should be on a new line.](https://www.codacy.com/app/ryotan/code-enum-support/file/2818755868/issues/source?bid=3375378&fileBranchId=3375797#l56)